src/util/std_code.h

```diff
@@ -236,6 +287,39 @@ class code_assignt:public codet
   {
     return op1();
   }
+
+  static void check(
+    const codet &code,
+    const validation_modet vm = validation_modet::INVARIANT)
+  {
+    DATA_CHECK(
+      code.operands().size() == 2, "assignment must have two operands");
+  }
+
+  static void validate(
+    const codet &code,
+    const namespacet &ns,
+    const validation_modet vm = validation_modet::INVARIANT)
+  {
+    check(code, vm);
+
+    DATA_CHECK(
+      base_type_eq(code.op0().type(), code.op1().type(), ns),
+      "lhs and rhs of assignment must have same type");
+  }
```

Comment:
It is no longer necessary to use base_type_eq to compare types -- you can simply use == !

Reply:
That's not quite the case just yet, is it? Otherwise we should have a major campaign removing all cases of `base_type_eq`.

Reply:
Yes, we should have that campaign, together with getting rid of most instances of ns.follow().